require('rollup/dist/loadConfigFile') causes unexpected ESM-related error

### Rollup Version

2.70.1

### Operating System (or Browser)

Linux

### Node Version (if applicable)

v14.17.5

### Link To Reproduction

N/A

### Expected Behaviour

In our project we build a watch mode in which we run `node build/watch.js` and in there run `const loadConfigFile = require('rollup/dist/loadConfigFile');` among other things. This has worked for years, and while the convention in ESM code is to include an extension, in CJS it tends to have been omitted.

### Actual Behaviour


In the last release, this seems to have stopped working. We now get an ESM error.. from a CJS import in a CJS file.

```
Uncaught Error: Cannot find module '/qunit/node_modules/rollup/dist/loadConfigFile'
    at createEsmNotFoundErr (internal/modules/cjs/loader.js:916:15)
    at finalizeEsmResolution (internal/modules/cjs/loader.js:909:15)
    at resolveExports (internal/modules/cjs/loader.js:449:14)
    at Function.Module._findPath (internal/modules/cjs/loader.js:489:31)
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:875:27)
    at Function.Module._load (internal/modules/cjs/loader.js:745:27)
    at Module.require (internal/modules/cjs/loader.js:961:19)
    at require (internal/modules/cjs/helpers.js:92:18) 
```

I've not included a test case as I believe this issue is already understood and discussed at https://github.com/rollup/rollup/pull/4436 and https://github.com/rollup/rollup/pull/4443. However that's quite burried and I'm hoping this will help others to find the issue quicker and avoid further duplicate reports.

**The short-term workaround is** to include `.js` in your require statement, or pin to an older version of Rollup.

```diff
@@ input @@
- const loadConfigFile = require('rollup/dist/loadConfigFile');
+ const loadConfigFile = require('rollup/dist/loadConfigFile.js');

@@ output @@
- Uncaught Error: Cannot find module '/qunit/node_modules/rollup/dist/loadConfigFile'
+ [AsyncFunction: loadAndParseConfigFile]
```

--- a/docs/02-javascript-api.md
+++ b/docs/02-javascript-api.md
@@ -260,7 +260,7 @@ See above for details on `inputOptions` and `outputOptions`, or consult the [big
 In order to aid in generating such a config, rollup exposes the helper it uses to load config files in its command line interface via a separate entry-point. This helper receives a resolved `fileName` and optionally an object containing command line parameters:
 
 ```js
-const loadConfigFile = require('rollup/dist/loadConfigFile');
+const loadConfigFile = require('rollup/loadConfigFile');
 const path = require('path');
 const rollup = require('rollup');
 

--- a/package.json
+++ b/package.json
@@ -136,6 +136,7 @@
       },
       "default": "./dist/es/rollup.browser.js"
     },
+    "./dist/loadConfigFile": "./dist/loadConfigFile.js",
     "./dist/*": "./dist/*"
   }
 }

--- a/package.json
+++ b/package.json
@@ -136,6 +136,7 @@
       },
       "default": "./dist/es/rollup.browser.js"
     },
+    "./loadConfigFile": "./dist/loadConfigFile.js",
     "./dist/loadConfigFile": "./dist/loadConfigFile.js",
     "./dist/*": "./dist/*"
   }

